torchgeo/datasets/sentinel.py

```diff
@@ -359,8 +361,10 @@ def __init__(
         """
         bands = bands or self.all_bands
         self.filename_glob = self.filename_glob.format(bands[0])
-        self.filename_regex = self.filename_regex.format(res)
-
+        if isinstance(res, float):
+            self.filename_regex = self.filename_regex.format(int(res))
+        else:
+            self.filename_regex = self.filename_regex.format(int(res[0]))
```

Comment:
Might be cleaner to convert res, then we don't have to duplicate the whole regex thing twice.

Reply:
One concern I have here is that if someone passes in a different res parameter to Sentinel2 then nothing will work because it won't find the data. That's a different problem though :)